widgets/weather_all.lua

```diff
@@ -58,6 +58,19 @@ local function parse(stdout, stderr, exitreason, exitcode)
     _weather["{press}"]   = -- Pressure in hPa
        string.match(stdout, "Pressure[%s].+%((.+)[%s]hPa%)") or _weather["{press}"]
 
+    local year, month, day, hour, min =
+        string.match(stdout, "(%d%d%d%d).(%d%d).(%d%d) (%d%d)(%d%d) UTC")
+
+    local utctable = {
+        year = year,
+        month = month,
+        day = day,
+        hour = hour,
+        min = min,
+    }
+
+    _weather["{when}"] = os.time(utctable)
```

Comment:
Please localize `os.time`, which might allow the code to run a bit faster. It might not matter much on current computers, but that's what I see in most Lua projects--we might be simply paranoid though.

Additionally, have you considered the case when stdout is abnormal, which could be due to server maintenance or wrong station code? If that happens, fallback to `N/A`.

On coding style, I'd prefer to have trailing newlines 63 and 71 removed, since this block of code forms a logical section. It's, however, mostly because the function is already tooo long (mea culpa) and thus blank line would be quite distracting.

Reply:
I'll take care of all of these; thanks for explaining some of them, I'd always wondered about the localization!

Reply:
The `N/A` fallback is a little tricky.  If we surface the information as seconds since the epoch, the user will have to check for `N/A` to guard against passing non-integers to `os.date` (or whatever.)  On the other hand, if we pass it as a zero, then the user has to check for a zero to render it as `N/A` themselves.  Ideas?

Reply:
As you can see at the beginning of `parse`, `_weather` was initiated with `N/A`. I'm just trying to follow this convention. IMHO if users gotta check, then `nil` would be most Lua-ish, so I guess it was because both numbers and `N/A` can be lazily and prettily formatted with `%s`.